spring-cloud-stream/src/test/java/org/springframework/cloud/stream/binder/InputOutputBindingOrderTest.java

```diff
@@ -46,15 +42,11 @@
  */
 public class InputOutputBindingOrderTest {
 
-	@Before
-	public void before() {
-		System.setProperty("spring.main.allow-bean-definition-overriding", "true");
-	}
-
 	@SuppressWarnings({"rawtypes", "unchecked"})
 	@Test
 	public void testInputOutputBindingOrder() {
-		ConfigurableApplicationContext applicationContext = SpringApplication.run(TestSource.class, "--server.port=-1");
+		ConfigurableApplicationContext applicationContext = SpringApplication.run(TestSource.class, "--server.port=-1",
```

Comment:
The wrong web exclude property again.
I mean there is no guarantee that such a `server.port=-1` turn off web environment in your application context.